backend/src/main/java/com/bakdata/conquery/resources/api/ConceptsProcessor.java

```diff
@@ -191,13 +191,12 @@ public AutoCompleteResult autocompleteTextFilter(SelectFilter<?> filter, Optiona
 	private List<FEValue> listAllValues(SelectFilter<?> filter) {
 		final Namespace namespace = namespaces.get(filter.getDataset().getId());
 
-		List<FEValue> out = new ArrayList<>();
-
-		for (TrieSearch<FEValue> search : namespace.getFilterSearch().getSearchesFor(filter)) {
-			out.addAll(search.listItems());
-		}
-
-		return out;
+		return namespace.getFilterSearch().getSearchesFor(filter)
+						.stream()
+						.map(TrieSearch::listItems)
+						.flatMap(Collection::stream)
+						.distinct()
+						.collect(Collectors.toList());
```

Comment:
kann man die Searches eines Filters nicht in einen Tree packen, damit nicht immer Distinct aufgerufen werden muss?

Reply:
Klar könnte man alle Werte hinter einem Filter in einen Baum packen, dann wären die aber sehr häufig mehrfach gespeichert (mehrere Suchen in unterschiedlichen Filtern über die gleiche Spalte - ggf unterschiedlich Parametrisiert) was ich hiermit vermieden habe, dazu erlaubt das auch eben unkompliziert das runterpriorisieren von unspezifischeren Werten.